stan/math/prim/fun/svd_U.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef STAN_MATH_PRIM_FUN_SVD_U_HPP
+#define STAN_MATH_PRIM_FUN_SVD_U_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Given input matrix m, return matrix U where `m = UDV^{T}`
+ *
+ * @tparam EigMat type of the matrix
+ * @param m MxN input matrix
+ * @return Orthogonal matrix U
+ */
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_not_vt_var<EigMat>* = nullptr>
+Eigen::Matrix<value_type_t<EigMat>, Eigen::Dynamic, Eigen::Dynamic> svd_U(
+    const EigMat& m) {
+  check_nonzero_size("svd_U", "m", m);
```

Comment:
@SteveBronder @andrjohns I'm struggling on this one -- at this point most of the matrix code accepts matrices of size zero and doesn't throw? Is that what we want in the future? We could definitely make this function return 0x0 matrices and not error.

Or I guess in general when should the matrix code throw and when should it just accept size zero?matrices?